src/thunderbots/software/test/ai/hl/stp/tactic/tactic.cpp

```diff
@@ -131,3 +131,183 @@ TEST(TacticTest, test_tactic_restarts_when_set_to_loop_infinitely)
         EXPECT_FALSE(tactic.done());
     }
 }
+
+
+
+typedef std::tuple<std::vector<RefboxGameState>, std::vector<AvoidArea>>
+    AvoidAreasTestParams;
+
+
+AvoidAreasTestParams makeParams(std::vector<RefboxGameState> refbox_states,
+                                std::vector<AvoidArea> avoid_areas)
+{
+    return std::make_tuple<std::vector<RefboxGameState>, std::vector<AvoidArea>>(
+        std::move(refbox_states), std::move(avoid_areas));
+}
+
+class TacticAvoidAreasTest : public ::testing::TestWithParam<AvoidAreasTestParams>
+{
+};
+
+TEST_P(TacticAvoidAreasTest, test_default_avoid_areas)
+{
+    std::vector<RefboxGameState> refbox_game_states = std::get<0>(GetParam());
+    std::vector<AvoidArea> expected_avoid_areas     = std::get<1>(GetParam());
+
+    Robot robot = Robot(0, Point(1, 1), Vector(), Angle::zero(), AngularVelocity::zero(),
+                        Timestamp::fromSeconds(0));
+    MoveTestTactic tactic = MoveTestTactic(true);
+    tactic.updateRobot(robot);
+
+    World world;
+    for (auto refbox_game_state : refbox_game_states)
+    {
+        world.mutableGameState().updateRefboxGameState(refbox_game_state);
+    }
+
+    auto next_intent = tactic.getNextIntent(world);
+    ASSERT_TRUE(next_intent);
+
+    std::string refbox_states_str;
+    for (auto state : refbox_game_states)
+    {
+        std::stringstream ss;
+        ss << state << ", ";
+        refbox_states_str += ss.str();
+    }
+
+    EXPECT_EQ(expected_avoid_areas, next_intent->getAreasToAvoid())
+        << "Failed with refbox states: " << refbox_states_str;
+}
+
+// Parameterized test to check that we return the correct areas to avoid for a
+// given refbox state
+INSTANTIATE_TEST_CASE_P(
+    All, TacticAvoidAreasTest,
+    ::testing::Values(
+        // Halt
+        makeParams({RefboxGameState::HALT},
+                   {AvoidArea::HALF_METER_AROUND_BALL, AvoidArea::ENEMY_DEFENSE_AREA,
+                    AvoidArea::FRIENDLY_DEFENSE_AREA}),
+        // Stop
+        makeParams({RefboxGameState::STOP}, {AvoidArea::HALF_METER_AROUND_BALL,
+                                             AvoidArea::INFLATED_ENEMY_DEFENSE_AREA,
+                                             AvoidArea::FRIENDLY_DEFENSE_AREA}),
+        // Our kickoff setup
+        makeParams({RefboxGameState::PREPARE_KICKOFF_US},
+                   {AvoidArea::ENEMY_DEFENSE_AREA, AvoidArea::FRIENDLY_DEFENSE_AREA}),
+        // Our kickoff
+        makeParams({RefboxGameState::PREPARE_KICKOFF_US, RefboxGameState::NORMAL_START},
+                   {AvoidArea::ENEMY_DEFENSE_AREA, AvoidArea::FRIENDLY_DEFENSE_AREA}),
+        // Their kickoff setup
+        makeParams({RefboxGameState::PREPARE_KICKOFF_THEM},
```

Comment:
Should also avoid enemy half of field

Reply:
This whole test got revamped, please re-review the whole thing. This specifically should be good though.